ykbh/src/lib.rs

```diff
@@ -0,0 +1,273 @@
+    }
+
+    /// Get the pointer to a Local.
+    fn local_ptr(&self, local: &Local) -> *mut u8 {
+        let offset = self.offsets[usize::try_from(local.0).unwrap()];
+        unsafe { self.locals.add(offset) }
+    }
+
+    /// Get the pointer for an IPlace, while applying all offsets.
+    fn iplace_to_ptr(&self, place: &IPlace) -> *mut u8 {
+        match place {
+            IPlace::Val {
+                local,
+                off,
+                ty: _ty,
+            } => {
+                // Get a pointer to the Val.
+                let dest_ptr = self.local_ptr(&local);
+                unsafe { dest_ptr.add(usize::try_from(*off).unwrap()) }
+            }
+            IPlace::Indirect { ptr, off, ty: _ty } => {
+                // Get a pointer to the Indirect.
+                let dest_ptr = self.local_ptr(&ptr.local) as *mut usize;
+                // Dereference the pointer by reading its value.
+                let val = unsafe { std::ptr::read::<usize>(dest_ptr) };
+                // Add the offsets of the Indirect.
+                let ptr = val + usize::try_from(ptr.off).unwrap() + usize::try_from(*off).unwrap();
+                // Now return the value as a pointer.
+                ptr as *mut u8
+            }
+            _ => unreachable!(),
+        }
+    }
+
+    /// Write some bytes to an IPlace. The amount of bytes is determined by the type of the
+    /// destination.
+    fn write(&mut self, dest: &IPlace, src: *const u8) {
+        match dest {
+            IPlace::Val {
+                local: _,
+                off: _,
+                ty,
+            }
+            | IPlace::Indirect { ptr: _, off: _, ty } => {
+                let size = usize::try_from(SIR.ty(ty).size()).unwrap();
+                let ptr = self.iplace_to_ptr(dest);
+                unsafe {
+                    std::ptr::copy(src, ptr, size);
+                }
+            }
+            _ => unreachable!(),
+        }
+    }
+
+    /// Implements the Store statement.
+    fn store(&mut self, dest: &IPlace, src: &IPlace) {
+        match src {
+            IPlace::Val { .. } | IPlace::Indirect { .. } => {
+                let src_ptr = self.iplace_to_ptr(src);
+                self.write(dest, src_ptr);
+            }
+            IPlace::Const { val, ty } => {
+                self.store_const(dest, val, ty);
+            }
+            _ => todo!(),
+        }
+    }
+
+    /// Writes a constant to an IPlace.
+    fn store_const(&mut self, dest: &IPlace, val: &Constant, ty: &TypeId) {
+        match val {
+            Constant::Int(ci) => match ci {
+                ConstantInt::UnsignedInt(ui) => match ui {
+                    UnsignedInt::U8(v) => self.write(dest, [*v].as_ptr()),
+                    _ => todo!(),
+                },
+                ConstantInt::SignedInt(_si) => todo!(),
+            },
+            Constant::Bool(_b) => todo!(),
+            Constant::Tuple(t) => {
+                if SIR.ty(t).size() == 0 {
```

Comment:
Wouldn't this be clearer as `if SIR.ty(t).size() != 0 { todo!(); }`?

Reply:
I wasn't sure if we might need to do something someday with ZSTs, but am happy to change it.